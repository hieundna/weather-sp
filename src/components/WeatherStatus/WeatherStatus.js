import '../../styles/WeatherStatus.scss';
import Context from '../../contexts/context'
import { useContext } from 'react';

function WeatherStatus() {
    const {weather} = useContext(Context);
    return (
        <div className='main'>
            <div className='state-icon'>
                {weather ? 
                    <img src={`http://openweathermap.org/img/wn/${weather?.weather[0]?.icon}@2x.png`} alt='cloud-icon' /> 
                : <i className="fas fa-cloud fa-3x"></i> }
            </div>
            <div className='main-title'>
                <div className='state'>{weather?.weather[0]?.main || 'Not found'}</div>
                <div className='detail'>
                    <i className="fas fa-thermometer-half"></i>
                    <span>{weather?.main?.temp || '--'}Â°C</span>
                </div>
            </div>
            <div className='detail'>
                <i className="far fa-tint"></i>
                <span>{weather?.main?.humidity || '--'}%</span>
            </div>
        </div>
    );
}

export default WeatherStatus;
