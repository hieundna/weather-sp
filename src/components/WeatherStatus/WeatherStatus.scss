.main {
    display: flex;
    justify-content: center;
    align-content: center;

    .state-icon {
        align-self: center;
    }

    .main-title {
        margin-left: 30px;
        margin-right: 30px;

        .state {
            font-size: 40px;
        }
    }

    .detail {
        font-size: 25px;
        margin-top: 15px;
        align-self: flex-end;

        i {
            margin-right: 10px;
        }
    }
}