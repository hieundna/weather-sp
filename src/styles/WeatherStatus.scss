.main {
    display: flex;
    justify-content: center;
    align-content: center;

    .state-icon {
        align-self: center;
    }

    .main-title {
        margin-left: 30px;
        margin-right: 30px;

        .state {
            font-size: 40px;
        }
    }

    .detail {
        font-size: 25px;
        margin-top: 15px;
        align-self: flex-end;

        i {
            margin-right: 10px;
        }
    }
}

@media screen and (max-height: 694px) {
    .main {
        .state {
            font-size: 30px !important;
        }
        .detail {
            font-size: 18px;
            margin-top: 7px;
        }
        .state-icon img{
            width: 65px;
        }
    }
}