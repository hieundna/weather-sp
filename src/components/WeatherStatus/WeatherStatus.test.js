import React from 'react';
import WeatherStatus from './WeatherStatus.js';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Context from '../../contexts/context';

const value = {
    weather: {
        weather: [
            {
                main: "Clouds",
                icon: "03n"
            }
        ],
        main: {
            temp: 28,
            feels_like: 32.57,
            humidity: 83
        }
    }
}

const wrapper = mount(<Context.Provider value={value}> <WeatherStatus /> </Context.Provider>);
describe('WeatherStatus Component', () => {
    test("expect to render WeatherStatus component", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const weatherStatusComponent = wrapper.find(".main");
        // expect(wrapper.props().weather).toEqual('Clouds');
        expect(weatherStatusComponent.length).toBe(1);
    });

})