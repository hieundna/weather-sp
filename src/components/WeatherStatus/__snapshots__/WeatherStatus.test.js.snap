// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherStatus Component expect to render WeatherStatus component 1`] = `
Array [
  " ",
  <WeatherStatus>
    <div
      className="main"
    >
      <div
        className="state-icon"
      >
        <img
          alt="cloud-icon"
          src="http://openweathermap.org/img/wn/03n@2x.png"
        />
      </div>
      <div
        className="main-title"
      >
        <div
          className="state"
        >
          Clouds
        </div>
        <div
          className="detail"
        >
          <i
            className="fas fa-thermometer-half"
          />
          <span>
            28
            Â°C
          </span>
        </div>
      </div>
      <div
        className="detail"
      >
        <i
          className="far fa-tint"
        />
        <span>
          83
          %
        </span>
      </div>
    </div>
  </WeatherStatus>,
  " ",
]
`;
